### PR TITLE
chore: Include .envrc to repo and a bunch of goodies

### PR DESCRIPTION
**Pull request by wtfzambo**
- Added task runner Taskfile.dev and basic Taskfile
- Bumped devenv.lock for devenv v1.1
- More informative pytest output

> [!NOTE]
> remember to upgrade devenv to 1.1 so devenv.lock is compatible

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    silent: true
+    cmd: devenv processes down
+
+  test:
+    desc: Run pytest
+    internal: false
+    silent: true
+    cmds:
+      - pytest || true
+      - rm -rf ./tmp
```
Comment (shahinism) [inline Taskfile.yml:32]: What's this file for?

Comment (wtfzambo) [reply Taskfile.yml:32]: command shortcuts. It's like Makefile but uses Task

Comment (shahinism) [reply Taskfile.yml:32]: devenv scripts and commands provide the same functionality without needing any extra dependency.

Comment (wtfzambo) [reply Taskfile.yml:32]: Oh, wasn't aware of that. I can change it

**Comment by wtfzambo** [comment]
Example output ouf 7b9cad9:
image

File: devenv.nix [hunk 2]
```diff
@@ -43,6 +43,36 @@ in
...
 
+  # convenient shortcuts
+  scripts.up.exec = "devenv up -d";
+  scripts.up.description = "Start processes in the background.";
+
+  scripts.down.exec = "devenv processes down";
+  scripts.down.description = "Stop processes.";
+
+  scripts.tests.exec = "pytest && rm -rf ./tmp";
+  scripts.tests.description = "Run pytest and cleanup.";
```
Comment (shahinism) [inline devenv.nix:54]: For `tests` it's better to just run `./tests.sh` as it carries all the tests. And can you elaborate on what is `./tmp`? If it's temporary let's put it in `/tmp` instead.

Comment (wtfzambo) [reply devenv.nix:54]: @shahinism `./tmp` is a folder that gets automatically created by current tests, look 
image.
It doesn't do anything bad, I just find it distracting to have it there hanging.

As for `./tests.sh`, had missed that, I'll update.

Comment (wtfzambo) [reply devenv.nix:54]: image
here an example of where that local `./tmp/` comes from.

Comment (wtfzambo) [reply devenv.nix]: Addressed in f7d5c14

**Pull request merged**